action.yml

```diff
@@ -34,6 +35,14 @@ inputs:
     description: Show hash values of files to be uploaded
     required: false
     default: 'false'
+  dry_run:
+    description: Do everything up to the final package upload
+    required: false
+    default: 'false'
+  oidc_audience:
+    description: The custom audience to request in the OIDC credential
+    required: false
+    default: 'pypi'
```

Comment:
I'm a little torn here, I think this workflow should probably automatically determine the audience for PyPI/TestPyPI without configuration rather than defaulting to `pypi`, but this should also be override-able to handle any future non-PyPI index.

Maybe the default is an empty string, which enables automatic configuration for indexes that the workflow knows about, otherwise fails?

Reply:
Maybe a wild idea, but what if we asked the index what its audience was? `GET https://pypi.org/_/oidc/audience`?

Reply:
That works for me -- I don't see any security reason _not_ to do that.

Reply:
Done!